packages/files/utils.js

```diff
@@ -0,0 +1,457 @@
+const busboy = require('busboy')
+const cuid = require('cuid')
+const { WriteStream } = require('fs-capacitor')
+const jwt = require('jsonwebtoken')
+const { validate: validateUUID } = require('uuid')
+const { z } = require('zod')
+
+const { FILE_RECORD_META_FIELDS } = require('@open-condo/files/schema/models')
+const { FileRecord } = require('@open-condo/files/schema/utils/serverSchema')
+const  { GQLError } = require('@open-condo/keystone/errors')
+const FileAdapter = require('@open-condo/keystone/fileAdapter/fileAdapter')
+const { getKVClient } = require('@open-condo/keystone/kv')
+
+const { ERRORS } = require('./errors')
+
+const DEFAULT_USER_HOUR_QUOTA = 100
+const DEFAULT_IP_HOUR_QUOTA = 100
+
+const AppClientSchema = z.object({
+    name: z.string().min(3).optional(),
+    modelNames: z.array(z.string()).min(1).optional(),
+    secret: z.string().min(8),
+}).strict()
+
+const AppClientsSchema = z.record(
+    z.string().regex(/^[a-zA-Z0-9-]+$/),
+    AppClientSchema
+)
+
+const AppConfigSchema = z.object({
+    clients: AppClientsSchema,
+    quota: z.object({
+        user: z.number().optional().default(DEFAULT_USER_HOUR_QUOTA),
+        ip: z.number().optional().default(DEFAULT_IP_HOUR_QUOTA),
+    }).optional().default({
+        user: DEFAULT_USER_HOUR_QUOTA,
+        ip: DEFAULT_IP_HOUR_QUOTA,
+    }),
+})
+
+function validateAndParseFileConfig (config) {
+    const { data, success, error } = AppConfigSchema.safeParse(config)
+    if (!success) {
+        console.error(error.issues[0]?.message || 'Invalid file upload config')
```

Comment:
Its better to use z.prettifyError(error) instead of this

Reply:
Good point, I'll refactor this one